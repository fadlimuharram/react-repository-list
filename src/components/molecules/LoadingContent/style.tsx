import styled from 'styled-components';

export const Wrapper = styled.div`
	.wrapperLoading {
		display: flex;
		align-items: center;
		flex-direction: column;
		width: 400px;
		height: 400px;
		background-color: #fff;
		box-shadow: 0 1rem 2rem rgb(0 0 0 / 15%);
		margin-top: 5rem;
		margin-right: 1rem;
		margin-left: 1rem;
	}

	.content {
		width: 100%;
		border: 1px solid white;
		flex-grow: 1;
		display: flex;
		flex-direction: column;
		padding: 20px;
		justify-content: space-between;
	}

	.stripe {
		border: 1px solid white;
		height: 20%;
		width: 100%;
		background-color: #babbbc;
	}

	.wrapperLoading.loading .img,
	.wrapperLoading.loading .stripe {
		animation: hintloading 2s ease-in-out 0s infinite reverse;
		-webkit-animation: hintloading 2s ease-in-out 0s infinite reverse;
	}

	@keyframes hintloading {
		0% {
			opacity: 0.2;
		}
		50% {
			opacity: 1;
		}
		100% {
			opacity: 0.2;
		}
	}

	@-webkit-keyframes hintloading {
		0% {
			opacity: 0.2;
		}
		50% {
			opacity: 1;
		}
		100% {
			opacity: 0.2;
		}
	}
`;
