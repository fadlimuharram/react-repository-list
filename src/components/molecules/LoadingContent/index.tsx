import React from 'react';
import { Wrapper } from './style';

interface IProps {
	testId?: string;
}

const LoadingContent: React.FC<IProps> = ({ testId = '' }) => {
	return (
		<Wrapper data-testid={`loading-content-${testId}`}>
			<div className="wrapperLoading loading">
				<div className="content">
					<div className="stripe small-stripe"></div>
					<div className="stripe medium-stripe"></div>
					<div className="stripe long-stripe"></div>
				</div>
			</div>
		</Wrapper>
	);
};

export default LoadingContent;
