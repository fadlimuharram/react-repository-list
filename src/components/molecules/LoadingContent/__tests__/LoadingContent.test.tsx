import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LoadingContent from '..';

afterEach(() => {
	cleanup();
});

test('should render NLoading Content', () => {
	render(<LoadingContent testId="1" />);
	const headingElm = screen.getByTestId('loading-content-1');
	expect(headingElm).toBeInTheDocument();
});

test('Match Loading Content Snapshot', () => {
	const tree = renderer.create(<LoadingContent testId="1" />).toJSON();

	expect(tree).toMatchSnapshot();
});
