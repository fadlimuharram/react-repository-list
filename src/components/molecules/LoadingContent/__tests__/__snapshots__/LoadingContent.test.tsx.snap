// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Loading Content Snapshot 1`] = `
<div
  className="sc-bdnxRM gTeMJY"
  data-testid="loading-content-1"
>
  <div
    className="wrapperLoading loading"
  >
    <div
      className="img-container"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="stripe small-stripe"
      />
      <div
        className="stripe medium-stripe"
      />
      <div
        className="stripe long-stripe"
      />
    </div>
  </div>
</div>
`;
